### ISSUE TITLE
AWS Dynamodb CPP SDK can't work with scylla alternator

### ISSUE THREAD

**Issue by zhangsw** [scylladb/scylla#6319]
I use dynamodb cpp sdk to operate scylla, scylla always returns "Unsupported operation".
Python sdk works fine. 

Scylla version: 4.0 rc1
dynamodb cpp sdk version: master 

Tcpdump result of list tables operation:

```
POST / HTTP/1.1
Accept: */*
content-length: 16
content-type: application/x-amz-json-1.0
host: 10.10.10.16:8016
user-agent: aws-sdk-cpp/1.7.322 Linux/3.10.0-327.el7.x86_64 x86_64 GCC/8.3.1
x-amz-api-version: 2012-08-10
x-amz-target: DynamoDB_20120810.ListTables

{
	"Limit":	50
}HTTP/1.1 400 Bad Request
Content-Type: application/json
Content-Length: 106
Server: Seastar httpd
Date: 29 Apr 2020 09:13:54 GMT

{"__type":"com.amazonaws.dynamodb.v20120810#UnknownOperationException","message":"Unsupported operation "}HTTP/ 404 Not Found
Content-Type: application/json
Content-Length: 37
Date: 29 Apr 2020 09:13:54 GMT
Server: Seastar httpd

{"message": "Not found", "code": 404}
```
Other operations, like create table, put item have same problems.
I use the same code to test dynamodb local, it works fine. 
Besides, it seems that scylla alternator doesn't support 100-continue header, which is enabled in cpp sdk by default.

**Comment by psarna** [comment]
Thanks, I'll look into it. Meanwhile, could you rerun this test with trace log level enabled in Scylla and share the logs? You can enable trace level logging for alternator via our REST API, e.g.:
```bash
curl -X POST http://localhost:10000/system/logger/alternator-executor?level=trace
```

**Comment by psarna** [comment]
Ok, I see what the problem is. The cpp driver sends `X-Amz-Target` header name in lowercase (`x-amz-target`, while we check for a case-sensitive `X-Amz-Target`. It actually goes down to a problem in Seastar's HTTP server: https://github.com/scylladb/seastar/issues/636

So, the proper fix should be inside Seastar, but perhaps I'll also consider creating a quick workaround for alternator in order to make it work with popular drivers.

**Comment by psarna** [comment]
I created a workaround here, so if you build Scylla from source, then you can try it out: https://github.com/scylladb/scylla/pull/6322

I'll also look at the issue from Seastar side in order to provide a proper fix.

**Comment by psarna** [comment]
I queued the seastar fix to 4.0 as well, as it's important for alternator drivers.

**Comment by nyh** [comment]
The Seastar patch is in 4.1, and backported to 4.0, so no futher backports are needed. Removing the "Backport candidate" label.
